model/tensorflow/dffml_model_tensorflow/dnnr.py

```diff
@@ -0,0 +1,323 @@
+            if not repo.features(self.features):
+                continue
+            ret_repos.append(repo)
+            for feature, results in repo.features(self.features).items():
+                x_cols[feature].append(np.array(results))
+        for feature in x_cols:
+            x_cols[feature] = np.array(x_cols[feature])
+        self.logger.info("------ Repo Data ------")
+        self.logger.info("x_cols:    %d", len(list(x_cols.values())[0]))
+        self.logger.info("-----------------------")
+        input_fn = tensorflow.estimator.inputs.numpy_input_fn(
+            x_cols, shuffle=False, num_epochs=1, **kwargs
+        )
+        return input_fn, ret_repos
+
+    async def train(self, sources: Sources):
+        """
+        Train on data submitted via classify.
+        """
+        input_fn = await self.training_input_fn(
+            sources,
+            batch_size=20,
+            shuffle=True,
+            epochs=self.parent.config.epochs,
+        )
+        self.model.train(input_fn=input_fn, steps=self.parent.config.steps)
+
+    async def accuracy(self, sources: Sources) -> Accuracy:
+        """
+        Evaluates the accuracy of our model after training using the input repos
+        as test data.
+        """
+        if not os.path.isdir(self.model_dir_path):
+            raise NotADirectoryError("Model not trained")
+        input_fn = await self.evaluate_input_fn(
+            sources, batch_size=20, shuffle=False, epochs=1
+        )
+        metrics = self.model.evaluate(input_fn=input_fn)
+        return Accuracy(1-metrics["loss"]) # 1 - mse
+
+    async def predict(self, repos: AsyncIterator[Repo]) -> AsyncIterator[Repo]:
+        """
+        Uses trained data to make a prediction about the quality of a repo.
+        """
+        
+        if not os.path.isdir(self.model_dir_path):
+            raise NotADirectoryError("Model not trained")
+        # Create the input function
+        input_fn, predict_repo = await self.predict_input_fn(repos)
+        # Makes predictions on 
+        predictions = self.model.predict(input_fn=input_fn)
+   
+        for repo, pred_dict in zip(predict_repo, predictions):
+            repo.predicted(float(pred_dict["predictions"]),float('nan')) # 0,arbitary number to match fun signature -> (value,confidence)
+           
+            yield repo
+
+
+@entry_point("tfdnnr")
+class DNNRegressionModel(Model):
+    """
+    Implemented using Tensorflow's DNNEstimator. Models are saved under the
+    ``directory`` in subdirectories named after the hash of their feature names and hidden layer config.
+
+    usage :
+        * label_name : column_name of target
+        * NOTE : if multiple models are trained then `model-hidden` should be passed to hash to the
+                    model directory
+
+    dffml train \
+          -model tfdnnr \
+          -model-epochs 300 \
+          -model-steps 2000 \
+          -model-label_name {$TARGET_COLUMN_NAME} \
```

Comment:
Can we make `label_name` be `predict`, to keep consistent with scikit models

Reply:
done